src/download.ts

```diff
@@ -1,13 +1,39 @@
 import { get, request, type RequestOptions } from 'https';
-import { format } from 'util';
 import * as Url from 'url';
 import { Starter } from './starters';
 import { HttpsProxyAgent } from 'https-proxy-agent';
 
-const STARTER_URL = 'https://github.com/%s/archive/main.zip';
+/**
+ * Build a URL to retrieve a starter template from a GitHub instance
+ *
+ * This function assumes that the starter will always be in a GitHub instance, as it returns a URL in string form that
+ * is specific to GitHub.
+ *
+ * @param starter metadata for the starter template to build a URL for
+ * @returns the generated URL to pull the template from
+ */
+export function getStarterUrl(starter: Starter): string {
+  return new URL(`${starter.repo}/archive/main.zip`, getGitHubUrl()).toString();
+}
+
+/**
+ * Retrieve the URL for the GitHub instance to pull the starter template from
+ *
+ * This function searches for the following environment variables (in order), using the first one that is found:
+ * 1. npm_config_stencil_self_hosted_url
+ * 2. stencil_self_hosted_url
```

Comment:
Nit - should we flip the order of these since the CLI arg takes precedence if both are set?

Reply:
Yeah - good call. In the original impl, had the `.npmrc` config taking precedence before realizing that was a bad idea 😆 done in d220812afddf5a7412c26a6aa249aa8ea297ec26